google-cloud-translate/lib/google/cloud/translate.rb

```diff
@@ -179,6 +189,16 @@ def self.default_credentials scope: nil
           Google::Cloud.configure.credentials ||
           Translate::Credentials.default(scope: scope)
       end
+
+      ##
+      # @private Resolve project.
+      def self.resolve_project_id project_id, credentials
+        # Always cast to a string
+        return project_id.to_s unless credentials.respond_to? :project_id
+
+        # Always cast to a string
+        project_id || credentials.project_id.to_s
```

Comment:
Given the conditional early return on line 197, I'm having trouble following the logic here that prefers to return `project_id` before `credentials.project_id`.

Reply:
Yeah, blame Rubocop for forcing such a bad layout. The reason it is this way is because we don't want to call the `#project_id` method if it doesn't exist, which is why the early return is there.